test/test_transaction.py

```diff
@@ -22,27 +29,38 @@
 class TestTransaction:  # pylint: disable=too-many-public-methods
     """Unittest for the pymodbus.transaction module."""
 
-    client = None
-    decoder = None
-    _tcp = None
-    _tls = None
-    _rtu = None
-    _ascii = None
-    _manager = None
-    _tm = None
+    client: ModbusBaseSyncClient
```

Comment:
I don't see a need for type hinting the class attributes here at all; they will be defined in `setup_method()`.  Just removing the lines here satisfies `mypy`.